site/app/controllers/EmailStatusController.php

```diff
@@ -15,34 +17,77 @@ public function __construct(Core $core) {
     }
 
     /**
-     * @Route("/courses/{_semester}/{_course}/email_status")
+     * @Route("/courses/{_semester}/{_course}/email_status", methods={"GET"})
      * @AccessControl(role="INSTRUCTOR")
-     * @return MultiResponse
+     * @return WebResponse
      */
     public function getEmailStatusPage() {
         $semester = $this->core->getConfig()->getSemester();
         $course = $this->core->getConfig()->getCourse();
-        $result = $this->core->getQueries()->getEmailStatusWithCourse($semester, $course);
-        return MultiResponse::webOnlyResponse(
-            new WebResponse(
+
+        $num_page = $this->core->getSubmittyEntityManager()->getRepository(EmailEntity::class)->getPageNum($semester, $course);
+
+        return new WebResponse(
+            EmailStatusView::class,
+            'showEmailStatusPage',
+            $num_page,
+            $this->core->buildCourseUrl(["email_status_page"])
+        );
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/email_status_page", methods={"GET"})
+     * @Route("/api/courses/{_semester}/{_course}/email_status_page", methods={"GET"})
+     * @AccessControl(role="INSTRUCTOR")
+     * @return array
+     */
+    public function getEmailStatusesByPage(): array {
```

Comment:
I am confused what this function is supposed to be returning. You return a Json response which contains HTML from the rendered Twig. You specify that it will return an array though. If this were to be a web and an API route then it should be strictly a normal json output. I think this should be the normal route which returns a WebResponse. If you want to add this to an API route then that should simply return a JSON containing information about the email statuses per page but not the HTML.